sdk/src/types/block/signature/ed25519.rs

```diff
@@ -193,12 +192,12 @@ pub(crate) mod dto {
     }
 
     impl TryFrom<Ed25519SignatureDto> for Ed25519Signature {
-        type Error = Error;
+        type Error = SignatureError;
 
         fn try_from(value: Ed25519SignatureDto) -> Result<Self, Self::Error> {
             Ok(Self::from_bytes(
-                prefix_hex::decode(&value.public_key).map_err(|_| Error::InvalidField("publicKey"))?,
-                prefix_hex::decode(&value.signature).map_err(|_| Error::InvalidField("signature"))?,
+                prefix_hex::decode(&value.public_key).map_err(|_| SignatureError::InvalidPublicKey)?,
+                prefix_hex::decode(&value.signature).map_err(|_| SignatureError::InvalidSignature)?,
```

Comment:
If these two variants are only used for Hex, I would maybe have an Hex suffix, bit vague otherwise